src/main.ts

```diff
@@ -1,21 +1,34 @@
 import { NestFactory } from '@nestjs/core';
-
-import * as helmet from 'helmet';
-
+import serverlessExpress from '@vendia/serverless-express';
+import helmet from 'helmet';
+import { Callback, Context, Handler } from 'aws-lambda';
 import { AppModule } from './app.module';
 
 const port = process.env.PORT || 4000;
 
+let server: Handler;
+
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
+  await app.init();
 
   app.enableCors({
     origin: (req, callback) => callback(null, true),
   });
   app.use(helmet());
 
-  await app.listen(port);
+  const expressApp = app.getHttpAdapter().getInstance();
+  return serverlessExpress({ app: expressApp });
 }
 bootstrap().then(() => {
   console.log('App is running on %s port', port);
 });
```

Comment:
Was thinking about it. It is nice to have when developing localy, But idk how AWS handling this. Is your lambda running till timeout because of this?
Its like: Making request -> script start this part of code -> 30 sec timeout -> app.close()
I'm not sure about this, asking if you made any research about this.